core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -26,74 +34,97 @@
      */
     private final String identifier;
     private final Map<String, AmbassadorContainer> ambassadorContainers = new HashMap<>();
+    private final File composeFile;
     private Set<String> spawnedContainerIds;
+    private Map<String, Integer> scalingPreferences = new HashMap<>();
+    private DockerClient dockerClient;
+
+    /**
+     * Properties that should be passed through to all Compose and ambassador containers (not
+     * necessarily to containers that are spawned by Compose itself)
+     */
+    private Map<String, String> env = new HashMap<>();
+
+    private static final RateLimiter AMBASSADOR_CREATION_RATE_LIMITER = RateLimiterBuilder
+            .newBuilder()
+            .withRate(6, TimeUnit.MINUTES)
+            .withConstantThroughput()
+            .build();
 
     public DockerComposeContainer(File composeFile) {
-        this(composeFile, "up -d");
+        this(composeFile, Base58.randomString(6).toLowerCase());
     }
 
     @SuppressWarnings("WeakerAccess")
-    public DockerComposeContainer(File composeFile, String command) {
-        super("dduportal/docker-compose:1.6.0");
-
-        // Create a unique identifier and tell compose
-        identifier = Base58.randomString(6).toLowerCase();
-        addEnv("COMPOSE_PROJECT_NAME", identifier);
+    public DockerComposeContainer(File composeFile, String identifier) {
+        this.composeFile = composeFile;
 
-        // Map the docker compose file into the container
-        addEnv("COMPOSE_FILE", "/compose/" + composeFile.getAbsoluteFile().getName());
-        addFileSystemBind(composeFile.getAbsoluteFile().getParentFile().getAbsolutePath(), "/compose", READ_ONLY);
+        // Use a unique identifier so that containers created for this compose environment can be identified
+        this.identifier = identifier;
 
-        // Ensure that compose can access docker. Since the container is assumed to be running on the same machine
-        //  as the docker daemon, just mapping the docker control socket is OK.
-        // As there seems to be a problem with mapping to the /var/run directory in certain environments (e.g. CircleCI)
-        //  we map the socket file outside of /var/run, as just /docker.sock
-        addFileSystemBind("/var/run/docker.sock", "/docker.sock", READ_WRITE);
-        addEnv("DOCKER_HOST", "unix:///docker.sock");
-
-        if (command != null) {
-            setCommand(command);
-        }
+        this.dockerClient = DockerClientFactory.instance().client();
     }
 
-    @Override
-    public void start() {
-
+    @Override @VisibleForTesting
+    public void starting(Description description) {
         final Profiler profiler = new Profiler("Docker compose container rule");
         profiler.setLogger(logger());
         profiler.start("Docker compose container startup");
 
+        applyScaling(); // scale before up, so that all scaled instances are available first for linking
```

Comment:
Though non-intuitive, we do the equivalent of `docker-compose scale ...` before `docker-compose up` so that all instances are available for links.